service/mapper/mapper.go

```diff
@@ -464,34 +477,36 @@ Outer:
 			// We then divide the remaining paths into chunks of 1000. For each
 			// batch, we retrieve the payloads from the state trie as it was at
 			// the end of this block and index them.
-			count := 0
-			n := 1000
-			total := ((len(paths) + n - 1) / n)
-			log.Debug().Int("num_paths", len(paths)).Int("num_batches", total).Msg("path batching executed")
-			for start := 0; start < len(paths); start += n {
-				// This loop may take a while, especially for the root checkpoint
-				// updates, so check if we should quit.
-				select {
-				case <-m.stop:
-					break Outer
-				default:
-					// keep going
+			if m.cfg.IndexPayloads {
```

Comment:
What do you think of inverting this, e.g.:

```go
if !m.cfg.IndexPayloads {
	commit = step.Commit
	continue
}

// rest of work here
```

We will repeat `commit = step.Commit` at the end but we do have one level of nesting less.